### PR TITLE
Refactor RAFT command applications and handle resulting errors

### PR DESCRIPTION
**Pull request by redouan-rhazouani**
Previously the errors were ignored

 Added unit tests for applying commands

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: cloud/transport/cluster.go [hunk 1]
```diff
@@ -107,7 +107,7 @@ func (c *Cluster) serve() error {
 	c.grpcServer = grpc.NewServer()
 	cmd.RegisterClusterServiceServer(c.grpcServer, c)
 	if err := c.grpcServer.Serve(c.ln); err != nil {
-		c.log.Error("serving incoming requests", "error", err)
+		c.log.Error("serving incoming requests: " + err.Error())
```
Comment (reyreaud-l) [inline cloud/transport/cluster.go:110]: Side suggestion: would it make sense here to follow a more "structured" log approach where we would have the error in a dedicated json field ?

File: cloud/store/db.go [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+func (db *localDB) AddClass(cmd *command.ApplyRequest, nodeID string, schemaOnly bool) error {
+	req := command.AddClassRequest{}
+	if err := json.Unmarshal(cmd.SubCommand, &req); err != nil {
+		return fmt.Errorf("%w: %w", errBadRequest, err)
+	}
+	if req.State == nil {
+		return fmt.Errorf("%w: nil sharding state", errBadRequest)
+	}
+	if err := db.parser.ParseClass(req.Class); err != nil {
+		return fmt.Errorf("%w :parsing class: %w", errBadRequest, err)
```
Comment (reyreaud-l) [inline cloud/store/db.go:50]: Nitpick `"%w: parsing class: %w"`

File: cloud/store/raft.go [hunk 3]
```diff
@@ -129,119 +128,67 @@ func (st *Store) Open(ctx context.Context) (err error) {
...
-			return fmt.Errorf("nil sharding state")
-		}
-		if err := st.parser.ParseClass(req.Class); err != nil {
-			return Response{Error: err}
-		}
-		req.State.SetLocalName(st.nodeID)
-		if ret.Error = st.schema.addClass(req.Class, req.State); ret.Error == nil && !schemaOnly {
-			st.db.AddClass(req)
-		}
+		ret.Error = st.db.AddClass(&cmd, st.nodeID, schemaOnly)
```
Comment (reyreaud-l) [inline cloud/store/raft.go:156]: Great change, a lot simpler now to follow the logic

File: cloud/store/store.go [hunk 4]
```diff
@@ -35,18 +35,20 @@ var (
 	ErrNotOpen        = errors.New("store not open")
 )
 
-type DB interface {
-	AddClass(pl cmd.AddClassRequest) error
-	UpdateClass(req cmd.UpdateClassRequest) error
+// Indexer interface is employed to update both the collection and its indexers in the filesystem.
+// This corresponds to updating metadata for collections
+type Indexer interface {
```
Comment (reyreaud-l) [inline cloud/store/store.go:40]: Based on the comment above, wouldn't `MetadataHandler` or `MetadataUpdater` be a better name than just `Indexer` ? It seems that it both updates the metadata and the index for collections.

File: cloud/store/store.go [hunk 5]
```diff
@@ -89,9 +91,7 @@ type Store struct {
 
 	nodeID   string
 	host     string
-	schema   *schema
-	db       DB
-	parser   Parser
+	db       localDB
```
Comment (reyreaud-l) [inline cloud/store/store.go:94]: We call both `DB` the localDB and the indexer in different part of the package. Could we get a better name to properly differentiate them and avoid confusion when reading the code which one is doing which ?

**Review by reyreaud-l — COMMENTED** [review]
Overall LGTM. Great change to simplify the command dispatch. I think there are a few naming/clarification updates that could be beneficial.

File: cloud/transport/cluster.go (hunk 1 above)
Comment (redouan-rhazouani) [reply cloud/transport/cluster.go:110]: As the logging is already structured, this is simply about enhancing the error context

Comment (redouan-rhazouani) [reply cloud/transport/cluster.go:110]: As the logging is already structured, this is simply about enhancing the error context

File: cloud/store/store.go (hunk 5 above)
Comment (redouan-rhazouani) [reply cloud/store/store.go:94]: We only have localDB, DB doesn't exist anymore

File: cloud/store/store.go (hunk 4 above)
Comment (redouan-rhazouani) [reply cloud/store/store.go:40]: This is not related to metadata. I've changed the definition to be clearer

**Review by reyreaud-l — APPROVED** [review]

**Pull request merged**